ui/src/app/pages/analysis/runtime-panel.spec.tsx

```diff
@@ -159,6 +160,71 @@ describe('RuntimePanel', () => {
     expect(runtimeApiStub.runtime.gceConfig.machineType).toEqual('n1-standard-4');
   });
 
+  it('should create runtime with preset values instead of getRuntime values if configurationType is GeneralAnalysis', async() => {
+    // In the case where the user's latest runtime is a preset (GeneralAnalysis in this case)
+    // we should ignore the other runtime config values that were delivered with the getRuntime response
+    // and instead, defer to the preset values defined in runtime-presets.ts when creating a new runtime
+
+    const runtime = {...runtimeApiStub.runtime,
+      status: RuntimeStatus.Deleted,
+      configurationType: RuntimeConfigurationType.GeneralAnalysis,
+      gceConfig: {
+        ...defaultGceConfig(),
+        machineType: 'n1-standard-16',
+        diskSize: 1000
+      },
+      dataprocConfig: null
+    };
+    runtimeApiStub.runtime = runtime;
+    runtimeStore.set({runtime: runtime, workspaceNamespace: workspaceStubs[0].namespace});
+
+    const wrapper = await component();
+    await mustClickButton(wrapper, 'Create');
+
+    expect(runtimeApiStub.runtime.status).toEqual('Creating');
+    expect(compareFields(
+      runtimeApiStub.runtime.gceConfig,
+      runtimePresets.generalAnalysis.runtimeTemplate.gceConfig,
+      ['machineType', 'diskSize']
+    )).toBeTruthy();
+  });
+
+  it('should create runtime with preset values instead of getRuntime values if configurationType is HailGenomicsAnalysis', async() => {
+    // In the case where the user's latest runtime is a preset (HailGenomicsAnalysis in this case)
+    // we should ignore the other runtime config values that were delivered with the getRuntime response
+    // and instead, defer to the preset values defined in runtime-presets.ts when creating a new runtime
+
+    const runtime = {...runtimeApiStub.runtime,
+      status: RuntimeStatus.Deleted,
+      configurationType: RuntimeConfigurationType.HailGenomicAnalysis,
+      gceConfig: null,
+      dataprocConfig: {
+        ...defaultDataprocConfig(),
+        masterMachineType: 'n1-standard-16',
+        masterDiskSize: 999,
+        workerDiskSize: 444,
+        numberOfWorkers: 5
+      }
+    };
+    runtimeApiStub.runtime = runtime;
+    runtimeStore.set({runtime: runtime, workspaceNamespace: workspaceStubs[0].namespace});
+
+    const wrapper = await component();
+    await mustClickButton(wrapper, 'Create');
+
+    expect(runtimeApiStub.runtime.status).toEqual('Creating');
+    expect(compareFields(
+      runtimeApiStub.runtime.dataprocConfig,
+      runtimePresets.hailAnalysis.runtimeTemplate.dataprocConfig,
+      ['masterMachineType', 'masterDiskSize', 'workerDiskSize', 'numberOfWorkers']
+    )).toBeTruthy();
+  });
+
+  const compareFields = (a, b, fieldNames) => {
+    const pick = fp.flow(fp.pick(fieldNames));
+    return fp.isEqual(pick(a) , pick(b));
```

Comment:
nit: It would be better to use `expect().toEqual()` so that you get a useful error message when this fails. Maybe change the signature to `expectEqualForFields()` or `expectPartiallyEqual` and just do the expect in here